### ISSUE TITLE
Sympy has no Fourier transforms between continuous and discrete domains?

### ISSUE THREAD

**Issue by JohnStone2017** [sympy/sympy#14880]
If there is a way in Sympy, I have not been able to find it after searching.

Mathematica has a function FourierSequenceTransform, implementing the "Discrete-Time Fourier Transform". Its mathematical definition is shown as the 3rd item in the 'Details' section here:

http://reference.wolfram.com/language/ref/FourierSequenceTransform.html

That function's inverse has various names. Mathematica calls the inverse InverseFourierSequenceTransform; Julius Smith calls it "Fourier Series" here:

https://ccrma.stanford.edu/~jos/st/Fourier_Transforms_Continuous_Discrete_Time_Frequency.html

Sympy omits these functions, or they are included but I have been unable to find their names?

No Z-Transform in Sympy either?

**Comment by asmeurer** [comment]
See the current GSoC project https://github.com/sidhantnagpal/gsoc/wiki/GSoC-2018-Application-Sidhant-Nagpal:-Transforms,-Convolution-&-Linear-Recurrence-Evaluation. I don't know if it is planned to implement transforms on discrete expressions. CC @sidhantnagpal 

For Z transforms see https://github.com/sympy/sympy/issues/12502

**Comment by sidhantnagpal** [comment]
Discrete Time Fourier Transform (DTFT) and Z-Transform take input as infinite discrete-time signal.
DTFT produces a continuous frequency spectrum which is periodic. Z-transform also produces a continuous complex frequency spectrum.

As suggested by Aaron, I think both of these transforms depend on summing capabilities of SymPy.

DFT, on the other hand takes input as finite (or periodic for that matter) discrete-time signal and produces a sampled frequency spectrum.

**Comment by asmeurer** [comment]
Right. These things could be implemented today, but they would end up returning unevaluated Sum objects a large percentage of the time. 

So to make this useful we need a serious project to improve SymPy's summation capabilities (see https://github.com/sympy/sympy/wiki/GSoC-2018-Ideas#concrete-module-implement-karr-algorithm-a-decision-procedure-for-symbolic-summation, I'm assuming Karr is sufficient for most of these transforms but I could be wrong about that).

**Comment by sidhantnagpal** [comment]
Yes, I think this could be followed up in GSoC 2019, and correspondingly we can have DTFT, Z-Transforms implemented as well (it will be good to have them). I will be able to look into this post GSoC.

**Comment by JohnStone2017** [comment]
Thanks for the info.

**Comment by Naba7** [comment]
I am participating in GSoC-19,Can I start implementing Z-transforms? Please guide me the way to achieve this
